stack-graphs/src/partial.rs

```diff
@@ -2352,18 +2354,25 @@ impl PartialPath {
                 continue;
             }
             let mut new_path = self.clone();
+            let mut new_cycle_detector = path_cycle_detector.clone();
             // If there are errors adding this edge to the partial path, or resolving the resulting
             // partial path, just skip the edge — it's not a fatal error.
             if new_path.append(graph, partials, extension).is_err() {
                 continue;
             }
-            result.push(new_path);
+            if !new_cycle_detector.appended(graph, partials, extension.sink, &new_path) {
+                println!("cyclic {}", new_path.display(graph, partials));
```

Comment:
This should be guarded by `copious_debugging` or removed

Reply:
Should go!